kubernetes/vault-snapshot.sh

```diff
@@ -7,8 +7,22 @@ VAULT_TOKEN=$(vault write -field=token  auth/kubernetes/login role="${VAULT_ROLE
 export VAULT_TOKEN
 
 # create snapshot
-
 vault operator raft snapshot save /vault-snapshots/vault_"$(date +%F-%H%M)".snapshot
 
 # upload to s3
 s3cmd put /vault-snapshots/* "${S3_URI}" --host="${S3_HOST}" --host-bucket="${S3_BUCKET}"
+
+# remove expired snapshots
+if [ "${S3_EXPIRE_DAYS}" ]; then
+    s3cmd ls "${S3_URI}" --host="${S3_HOST}" --host-bucket="${S3_BUCKET}" | while read -r line; do
```

Comment:
There is a really strange behavior of `s3cmd` on my machine. When I test without setting `--access_key=$AWS_ACCESS_KEY_ID --secret_key=$AWS_SECRET_ACCESS_KEY` explicitly, the command fails with `403 (AccessDenied)` (even though I have the variables in the environment). Not quite sure why it appears to be working inside this container though..

Reply:
Maybe locally it's trying to get the credentials from `~/.aws` instead of the environment variables, but honestly I would assume that every tool should give environment variables the highest priority.

Reply:
I did not go down this rabbit hole to check it further :rabbit: However, it seems to work fine inside the CronJob, which is the main objective :smile: